app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -1,22 +1,174 @@
 package com.realityexpander.tasky.presentation.login_screen
 
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Button
+import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Text
-import androidx.compose.material.TextField
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.realityexpander.tasky.presentation.components.EmailField
+import com.realityexpander.tasky.presentation.components.PasswordField
+import com.realityexpander.tasky.presentation.destinations.RegisterScreenDestination
+import kotlinx.coroutines.launch
 
 @Composable
 @Destination(start = true)
 fun LoginScreen(
-    viewModel: LoginViewModel = hiltViewModel()
+    email: String? = null,
+    password: String? = null,
+    confirmPassword: String? = null,
+    navigator: DestinationsNavigator,
+    viewModel: LoginViewModel = hiltViewModel(),
 ) {
+    BackHandler(true) { /* We want to disable back clicks */ }
 
-    Text(text = "Login")
+    val loginState by viewModel.loginState.collectAsState()
+    val scope = rememberCoroutineScope()
+    val focusManager = LocalFocusManager.current
+
+    fun performLogin() {
+        scope.launch {
+            viewModel.onEvent(LoginEvent.ValidateEmail(loginState.email))
+            viewModel.onEvent(LoginEvent.ValidatePassword(loginState.password))
+            viewModel.onEvent(LoginEvent.Login(loginState.email, loginState.password))
```

Comment:
*Personal opinion alert*
To me, it should rather be the job of the ViewModel to decide what needs to be done in which order. When the user clicks on Login, all the UI knows is that exact action. The ViewModel could then validate the fields before attempting to make the network call. 
But, if you disagree with this, feel free to leave this, just sharing my view here.

Reply:
I can see the value in this.

OK, i will move this to the viewmodel and do the validations before the `LoginEvent.login`